### DISCUSSION TITLE
Dynamic import syntax (solid-app-router)

### DISCUSSION THREAD

**Discussion by tslocke** [solidjs/solid#859] [Q&A] [answered]
Can anyone explain what's going on here with the brackets inside the import string?

(from https://github.com/solidjs/solid-app-router)

```
const UserNotFound = lazy(() => import("/pages/users/[id]/[...all].js"));
```

Thanks

**Comment by ryansolid** [comment, marked as answer]
Oh that's just next.js style naming scheme. I was using that when figuring out for file based router and copied the example. The files can be named anything you want them to be.

**Reply by tslocke** [reply]
I feel like I must be missing something. I figured there was some import magic going on here, but if not, this example assumes a file literally named `[...all].js` and a directory named `[id]`. Is that intended?

**Reply by ryansolid** [reply]
Yes. It just had files/folders named those names. They could be named anything. I can see how that could be confusing as a base example since usually those file names do magic things. I literally just pulled it off a demo I was playing with.

**Reply by tslocke** [reply]
OK I learned something - that's a pretty wild design decision in next.js, to use the file/dir names to auto-configure route params. Once you get used to it it's actually a great idea. Anything that reduces config boilerplate is a win.

In the solid-app-router docs you do mention "great for filesystem based routing" but this isn't the full next.js-like thing right? It's just convenient because you can do the config and the imports all in the same place.

**Reply by ryansolid** [reply]
Yes but we use it the same way as Next in our new starter SolidStart (Still alpha). https://github.com/solidjs/solid-start
